pkg/gvkcompare/gvkcompare.go

```diff
@@ -0,0 +1,276 @@
+package gvkcompare
+
+import (
+	"sort"
+	"strings"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/discovery"
+)
+
+// GVKCompare reconciles a GVKCompare object
+type GVKCompare struct {
+	Plan         *migapi.MigPlan
+	SrcDiscovery *discovery.DiscoveryClient
+	DstDiscovery *discovery.DiscoveryClient
+	SrcClient    dynamic.Interface
+}
+
+// Compare GVKs on both clusters, find unsupported GVRs
+// and check each plan source namespace for existence of unsupported GVRs
+func (r *GVKCompare) Compare() error {
+	gvDiff, err := r.compareGroupVersions()
+	if err != nil {
+		return errors.Wrap(err, "Unable to compare GroupVersions between clusters")
+	}
+
+	unsupportedGVRs, err := r.unsupportedServerResources(gvDiff)
+	if err != nil {
+		return errors.Wrap(err, "Unable to get unsupported resources for scrCluster")
+	}
+
+	err = r.collectNamespaceReport(unsupportedGVRs)
+	if err != nil {
+		return errors.Wrap(err, "Unable to evaluate GVR gaps for migrated resources")
+	}
+
+	return nil
+}
+
+func (r *GVKCompare) PrepareSourceDiscovery(c client.Client) error {
```

Comment:
Instead of Prepare__() methods, I think the idiomatic way to accomplish complex construction would be to use a `NewComparator() (*Comparator, error)` function.